hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-resourcemanager/src/main/java/org/apache/hadoop/yarn/server/resourcemanager/scheduler/capacity/AutoCreatedQueueTemplate.java

```diff
@@ -154,46 +148,26 @@ public void setTemplateEntriesForChild(CapacitySchedulerConfiguration conf,
    */
   private void setTemplateConfigEntries(CapacitySchedulerConfiguration configuration,
                                         QueuePath queuePath) {
-    ConfigurationProperties configurationProperties =
-        configuration.getConfigurationProperties();
-
-    List<String> queuePathParts = Lists.newArrayList(queuePath.iterator());
-
-    if (queuePathParts.size() <= 1 && !queuePath.isRoot()) {
-      // This is an invalid queue path
-      return;
-    }
-    int queuePathMaxIndex = queuePathParts.size() - 1;
-
-    // start with the most explicit format (without wildcard)
-    int wildcardLevel = 0;
-    // root can not be wildcarded
-    int supportedWildcardLevel = Math.min(queuePathMaxIndex,
-            configuration.getMaximumAutoCreatedQueueDepth(queuePath.getFullPath()));
-    // Allow root to have template properties
-    if (queuePath.isRoot()) {
-      supportedWildcardLevel = 0;
-    }
-
-    // Collect all template entries
-    while (wildcardLevel <= supportedWildcardLevel) {
-      String templateQueuePath = String.join(".", queuePathParts);
-      // Get all configuration entries with
-      // yarn.scheduler.capacity.<queuePath> prefix
-      Map<String, String> queueProps = configurationProperties
-          .getPropertiesWithPrefix(getQueuePrefix(templateQueuePath));
-
-      // Store template, parent-template and leaf-template properties
-      for (Map.Entry<String, String> entry : queueProps.entrySet()) {
-        storeConfiguredTemplates(entry.getKey(), entry.getValue());
+    if (!queuePath.isInvalid()) {
+      ConfigurationProperties configurationProperties =
+          configuration.getConfigurationProperties();
+
+      int maxAutoCreatedQueueDepth = configuration
+          .getMaximumAutoCreatedQueueDepth(queuePath);
+      List<QueuePath> wildcardedQueuePaths =
+          queuePath.getWildcardedQueuePaths(maxAutoCreatedQueueDepth);
+
+      for (QueuePath templateQueuePath: wildcardedQueuePaths) {
+        // Get all configuration entries with
+        // yarn.scheduler.capacity.<queuePath> prefix
+        Map<String, String> queueProps = configurationProperties
+            .getPropertiesWithPrefix(getQueuePrefix(templateQueuePath));
+
+        // Store template, parent-template and leaf-template properties
+        for (Map.Entry<String, String> entry : queueProps.entrySet()) {
+          storeConfiguredTemplates(entry.getKey(), entry.getValue());
+        }
```

Comment:
I can see the logic has been moved to QueuePath here.
Is this thoroughly tested with unit tests?